hibernate-core/src/main/java/org/hibernate/type/descriptor/java/InstantJavaDescriptor.java

```diff
@@ -95,7 +101,13 @@ public Instant fromString(String string) {
 
 		if ( Timestamp.class.isInstance( value ) ) {
 			final Timestamp ts = (Timestamp) value;
-			return ts.toInstant();
+			/*
+			 * Workaround for HHH-13266 (JDK-8061577).
+			 * Ideally we'd want to use ts.toInstant(), but this won't always work.
+			 * ts.toInstant() assumes the number of milliseconds since the epoch
+			 * means the same thing in Timestamp and Instant, but it doesn't, in particular before 1900.
+			 */
+			return ts.toLocalDateTime().atZone( ZoneId.systemDefault() ).toInstant();
```

Comment:
Note this will fail with an exception for some dates... For example february 29th, 1500 exists for `java.util` (which uses the Julian calendar at that point, so 1500 is a leap year), but it does not exist for `java.time` (which uses the proleptic Gregorian calendar at that point, so 1500 is not a leap year).

As far as I can tell, this could only happen if the user bypassed Hibernate ORM to add this date in their database. But still, it's a problem.

I suppose I could add a check to turn the 29th February to the 1st March. But this feels deeply wrong.
I could also re-implement `toLocalDateTime()` and switch to another strategy before the Julian/Gregorian calendar cut-off, but the Julian/Gregorian calendar cut-off happened at a different date at different places...

Or, we could accept that such dates are not handled. I just checked, and postgresql does not accept the 29th of February, 1500 as a valid date anyway:

```
hibernate_orm_test=# select '1500-02-28T12:00:00'::timestamp;
      timestamp      
---------------------
 1500-02-28 12:00:00
(1 row)
hibernate_orm_test=# select '1500-02-29T12:00:00'::timestamp;
ERROR:  date/time field value out of range: "1500-02-29T12:00:00"
LINE 1: select '1500-02-29T12:00:00'::timestamp
               ^
```

Reply:
I suppose we will have to live with it. Let's see if we get some reports and decide then.